opencodeblocks/graphics/view.py

```diff
@@ -134,6 +134,85 @@ def middleMouseButtonRelease(self, event: QMouseEvent):
         super().mouseReleaseEvent(event)
         self.setDragMode(QGraphicsView.DragMode.RubberBandDrag)
 
+    def centerView(self, x: float, y: float):
+        """ Move the view so that the position (x,y) is centered. """
+        hsb = self.horizontalScrollBar()
+        vsb = self.verticalScrollBar()
+        hsb.setValue(x * self.zoom - self.width() / 2)
+        vsb.setValue(y * self.zoom - self.height() / 2)
+
+    def getDistanceToCenter(self, x: float, y: float) -> Tuple[float]:
+        """ Return the vector from the (x,y) position given to the center of the view """
+        ypos = self.verticalScrollBar().value()
+        xpos = self.horizontalScrollBar().value()
+        return (
+            xpos - x * self.zoom + self.width() / 2,
+            ypos - y * self.zoom + self.height() / 2
+        )
+
+    def moveViewOnArrow(self, event: QKeyEvent) -> bool:
+        """
+            OCBView reaction to an arrow key being pressed.
+            Returns True if the event was handled.
+        """
+        # The focusItem has priority for this event
+        if self.scene().focusItem() is not None:
+            return False
+        if len(self.scene().selectedItems()) > 0:
+            return False
+
+        key_id = event.key()
+        items = self.scene().items()
+        code_blocks = [i for i in items if isinstance(i, OCBBlock)]
```

Comment:
blocks in a more general name, as text blocks and so on might be in there

Reply:
I will edit this in the pull request that resolves #28